server/src/test/java/io/crate/types/BooleanTypeTest.java

```diff
@@ -39,10 +39,22 @@ public void test_cast_text_to_boolean() {
         assertThat(BooleanType.INSTANCE.implicitCast("FALSE"), is(false));
         assertThat(BooleanType.INSTANCE.implicitCast("f"), is(false));
         assertThat(BooleanType.INSTANCE.implicitCast("F"), is(false));
+        assertThat(BooleanType.INSTANCE.implicitCast("no"), is(false));
+        assertThat(BooleanType.INSTANCE.implicitCast("NO"), is(false));
+        assertThat(BooleanType.INSTANCE.implicitCast("n"), is(false));
+        assertThat(BooleanType.INSTANCE.implicitCast("off"), is(false));
+        assertThat(BooleanType.INSTANCE.implicitCast("OFF"), is(false));
+        assertThat(BooleanType.INSTANCE.implicitCast("0"), is(false));
         assertThat(BooleanType.INSTANCE.implicitCast("true"), is(true));
         assertThat(BooleanType.INSTANCE.implicitCast("TRUE"), is(true));
         assertThat(BooleanType.INSTANCE.implicitCast("t"), is(true));
         assertThat(BooleanType.INSTANCE.implicitCast("T"), is(true));
+        assertThat(BooleanType.INSTANCE.implicitCast("YES"), is(true));
+        assertThat(BooleanType.INSTANCE.implicitCast("yes"), is(true));
+        assertThat(BooleanType.INSTANCE.implicitCast("y"), is(true));
```

Comment:
also `Y`?

Reply:
I adjusted the test cases to be more complete